components/shared/profile/ProfileHeader.tsx

```diff
@@ -1,29 +1,74 @@
 import React from "react";
 import Image from "next/image";
-import { getSession } from "@/lib/authOptions";
 
-const ProfileHeader = async () => {
-  const session = await getSession();
+import { FiMapPin } from "react-icons/fi";
+import { FaLink } from "react-icons/fa6";
+import { CiCalendar } from "react-icons/ci";
 
+import { redirect } from "next/navigation";
+import { IUser } from "@/database/models/user.model";
+import { format } from "date-fns";
+import { Button } from "@/components/ui/button";
+import Link from "next/link";
+import icnEdit from "@/public/icons/icn-edit.svg";
+interface ProfileHeaderProps {
+  user: Partial<IUser>;
+}
+const ProfileHeader = async ({ user }: ProfileHeaderProps) => {
+  if (!user) {
+    redirect("/sign-in");
+  }
   return (
-    <div className={`mt-[40px] flex h-[90px] items-center `}>
-      <div className={` flex-0 ml-[10px]`}>
+    <section
+      className={`mt-[40px] flex h-[90px] items-center justify-between gap-5`}
+    >
+      <div className={` flex-0 justify-start gap-5`}>
         <Image
           alt="pic"
           height={90}
           width={90}
-          src={session?.user?.image!}
+          src={user?.image!}
           className="rounded-[5px]"
         />
       </div>
-      <div className="ml-[10px] flex-1 flex-col gap-4">
-        <div className="flex-1">
-          <span className="display-2-bold text-left align-top text-white-100">
-            {session?.user?.name!}
-          </span>
+      <div className="flex-1  flex-col justify-end gap-2.5 space-y-8">
+        <span className="display-2-bold text-left align-top text-white-100">
+          {user?.fullname!}
+        </span>
+        <div className="flex w-1/2  flex-row items-start justify-between gap-2">
```

Comment:
```suggestion
        <div className="flex w-1/2  flex-row items-center justify-between gap-2">
```
This will line things up better